### PR TITLE
jobs: truncate all errors before persisting them

### PR DESCRIPTION
**Pull request by stevendanna**
In cockroachdb#73303 we truncated the row data in this error message to prevent
problems with large row data preventing the job status from being
saved.

However, truncating the row means that our
"experimental_save_rejected" option does not work as expected. This
feature previously saved entire rows and now it would have truncated
rows in some case.

Now, we only truncate the error when producing the error
string. Further, we also add truncation to the job system to prevent
other parts of the system from saving too much data to the job system.

Release note: None

### CODE REVIEW

**Comment by dt** [comment]
Hm. The error itself could have the un-parsable value in it, so I think we'd still want to assume the error message has user data in it or at least something that could reveal something about user data. Given that, I'm not sure that removing this wins as much? including the "row" that didn't parse has proved really helpful in catching incorrect record separators configuration before.

**Comment by stevendanna** [comment]
@dt I agree I'm pretty bummed about the loss of the row for spotting easy-to-spot problems in the data.  I suppose we could truncate it for the purposes of error reporting but _not_ for purposes of sending it to the record of rejected entries.

**Comment by dt** [comment]
We should probably truncate the whole error before we persist it at the jobs level honestly, since any number of error creators could create outsized errors (e.g. if an external storage implementation includes the body of non-2xx server replies some service returns giant xml errors), not just this wrap with the row content.

**Comment by stevendanna** [comment]
This should be good for another look. I added error truncation to the underlying job system.  I left the truncation for the row in import specifically because we log that in at least one place.

File: pkg/jobs/jobs_test.go [hunk 1]
```diff
@@ -1423,6 +1423,17 @@ func TestJobLifecycle(t *testing.T) {
...
+			const hugeErr = "boom: fmljqpftdixvfkcscrkhtezubzbwkxprfnqbtqttbvddvhzsokdvwiuimksfdnjsiwnozzavbsokpdravmskqqlolnvcedkwhlqpppcraljqkpregixfmspklrnihulsmsacelxfuexabroiqpsjeyraurbivdenhnluwiottbktrlnyijonilmcxddjvzcwatqqnsbquhoqaqdwtopubtcjtbyqinibryjgrcvrzqjhlpyjcqdjqfzouljmglflurgwfatyqnhwhjtssivtxpyrtgillgxiplnglvvwnohbmsnqoqtqzvhuofpwqxwjnpgbffmjkczxuwdloriijfsapqobctifgqrvuepojhpypkundijguqtrckhkiyuhlbfonzkhazyzipzyopsczqufxhgxlzcqcceawidfgimzsywdtefciuejpommuqwjmbhwanxsrodmrazrsjbwbrfrzcfzwcpliwpfpciyxptogiudscbgleoiqajhtcbqvwdezabulnkzbqmzyujqtnntxveiwtuwbcvbvhmspobclkonzxwqhowtyboolqbupovifchblxkqdjgnilpqekfufolwxmiskxihtocbrsdvndlzobqjtsqutargdjlhlrxptczkaicncnfepadwnqruigxtfqtymgsrfruakmniyqswenyqcwnwtnaeuivyohcbihtbwpfppochifbfdrcdaoijauhjgkneqvjkxkcxxiqqfhbgjfcduwudcnsmjblapdpwxoqpcblvagganrdmrmbmlooydekaadlnrcptfrketjrspktayqudgyhjeuufvqgnkvtobtszcgdqqkemcqsjsbrnvykwfoolcundzbkzbkhksxraqvbzevnuzgywtrqprcftddhjoqswlqbkxaqrjtdlbqqeznhipdsbvjltlbitqlmuxlmwmmimpezfsnyoycwhbfkfcmlypidhvczynbgnkpfzjvzuznlpjriscglwrymvagbefmdpcrwrhibystalngiusnqzvutqyxmxvoaqgyqrgudvozjvyqvrqjpyxirwwwacfzilurtjevbk"
```
Comment (dt) [inline pkg/jobs/jobs_test.go:1427]: I've sometimes found `strings.Repeat` handy for cases like this.

**Review by dt — APPROVED** [review]

**Pull request merged**